File: ruby/spec/locales/generated_messages.spec.rb
# frozen_string_literal: true

Enolib::Locales.constants.each do |locale_code|
  locale = Enolib::Locales.const_get(locale_code)

  describe locale do
    locale.constants.each do |name|
      describe name do
        translation = Enolib::Locales::En.const_get(name)

        if translation.is_a?(Proc)
          it 'contains a dynamic translated message generator function' do
            parameters = Array.new(translation.parameters.length)
            generated_message = translation.call(*parameters)
            expect(generated_message).to be_a(String)
          end
        else
          it 'contains a static string translation' do
            expect(translation).to be_a(String)
          end
        end
      end
    end
  end
end

File: javascript/specs/locales/generated_messages.spec.js
import { parse } from '../..';

import de from '../../locales/de.js';
import en from '../../lib/messages.js';
import es from '../../locales/es.js';

const locales = { de, en, es };

describe('Message locales', () => {
    for (const [locale, messages] of Object.entries(locales)) {
        describe(locale, () => {
            for (const [message, translation] of Object.entries(messages)) {
                describe(message, () => {
                    it('contains a static string translation or a message generator function', () => {
                        if(typeof translation === 'function') {
                            const generatedMessage = translation();
                            expect(typeof generatedMessage).toEqual('string');
                        } else {
                            expect(typeof translation).toEqual('string');
                        }
                    });
                });
            }
        });
    }
});
